src/runtime_src/core/pcie/tools/xbmgmt/xspi.cpp

```diff
@@ -1932,9 +1962,27 @@ static int bitstreamGuardAddress(pcidev::pci_device *dev, uint32_t& addr)
 
 int XSPI_Flasher::revertToMFG(void)
 {
+    int ret;
+    std::string errmsg;
+    uint16_t vendor;
+
+    // Vendor-dependent behaviour
+    mDev->sysfs_get<uint16_t>("", "vendor", errmsg, vendor, -1);
+    if (!errmsg.empty()) {
+        std::cout << errmsg << std::endl;
+        return -EINVAL;
+    }
+
+    switch (vendor) {
+        case ARISTA_ID:
+            return fullErase() ? 0 : -EINVAL;
```

Comment:
The concept of revertToMFG is to have a valid shell locked down on the flash always. This shell will take effect when there is no other valid shell on flash. When you erase the whole flash, do you still have a valid shell?

Reply:
In our case we do not have a golden image stored in the flash, the default state is an empty flash.

Reply:
> 
> 
> In our case we do not have a golden image stored in the flash, the default state is an empty flash.

OK, I see. Thanks.